File: Python/체육복.py
def solution(n, lost, reserve):    
    arr = []
    lost.sort()
    reserve.sort()
    tmp = lost[:]

    for i in tmp:
        if i in reserve:
            reserve.remove(i)
            lost.remove(i)

    for i in range (1, n + 1):
        arr.append(i)

    for i in lost:
        arr.remove(i)

    for i in reserve:
        if i - 1 not in arr:
            if i - 1 > 0:
                arr.append(i - 1)
        elif i not in arr:
            arr.append(i)
        elif i + 1 not in arr:
            if i + 1 <= n:
                arr.append(i + 1)
    print(arr)
    return len(arr)

File: Java/체육복.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution {
	public int solution(int n, int[] lost, int[] reserve) {
		int answer = n - lost.length;
		boolean[] arr = new boolean[n + 2];		
		ArrayList<Integer> lost_arr = new ArrayList<Integer>();

		Arrays.sort(lost);
		
		for (int i = 0; i <= n + 1; i++) {
			arr[i] = false;
		}
 
		for (int i = 0; i < reserve.length; i++) {
			arr[reserve[i]] = true;
		}

		for (int i = 0; i < lost.length; i++) {
			if (arr[lost[i]]) {
				answer += 1;
				arr[lost[i]] = false;
			} else {
				lost_arr.add(lost[i]);
			}
		}

		for (int i = 0; i < lost_arr.size(); i++) {
			int idx = lost_arr.get(i);
			if (arr[idx] == false) {
				if (arr[idx - 1]) {
					answer += 1;
					arr[idx - 1] = false;
				} else if (arr[idx + 1]) {
					answer += 1;
					arr[idx + 1] = false;
				}
			}
		}

		return answer;
	}
}